File: backend/src/main/java/com/commit/crm/model/Lead.java
package com.commit.crm.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "leads")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Lead {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @EqualsAndHashCode.Include
    private UUID id;

    @Column(name = "company_name", nullable = false, length = 200)
    private String companyName;

    @Column(length = 500)
    private String site;

    @Column(length = 200)
    private String instagram;

    @Column(length = 20)
    private String whatsapp;

    private String address;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 100)
    private LeadSegment segment;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "enriched_data", columnDefinition = "jsonb")
    private Object enrichedData;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    @Builder.Default
    private LeadStatus status = LeadStatus.NOVO;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by", nullable = false)
    private User createdBy;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "assigned_to")
    private User assignedTo;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: frontend/src/types/lead.ts
export enum LeadStatus {
  NOVO = 'NOVO',
  CONTATO = 'CONTATO',
  NEGOCIACAO = 'NEGOCIACAO',
  GANHO = 'GANHO',
  PERDIDO = 'PERDIDO',
  ARQUIVADO = 'ARQUIVADO',
}

export enum LeadSegment {
  TECNOLOGIA = 'TECNOLOGIA',
  FINANCAS = 'FINANCAS',
  SAUDE = 'SAUDE',
  EDUCACAO = 'EDUCACAO',
  VAREJO = 'VAREJO',
  OUTRO = 'OUTRO',
}

export interface Lead {
  id: string;
  companyName: string;
  site: string | null;
  instagram: string | null;
  whatsapp: string | null;
  address: string | null;
  segment: string;
  notes: string | null;
  status: string;
  createdByName: string;
  assignedToName: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface LeadRequest {
  companyName: string;
  site?: string;
  instagram?: string;
  whatsapp?: string;
  address?: string;
  segment: string;
  notes?: string;
}

export interface LeadListResponse {
  content: Lead[];
  page: number;
  size: number;
  totalElements: number;
  totalPages: number;
}
